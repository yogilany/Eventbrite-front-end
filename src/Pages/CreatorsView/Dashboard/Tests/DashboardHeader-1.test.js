const { shallow } = require("enzyme");
const { default: DashboardHeader } = require("../Components/DashboardHeader-1");

describe("DashboardHeader", () => {
  const tickets = 5;
  const sold = 3;
  const total = 10;

  it("renders correctly", () => {
    const wrapper = shallow(
      <DashboardHeader tickets={tickets} sold={sold} total={total} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders the TicketsCard component with the correct props", () => {
    const wrapper = shallow(
      <DashboardHeader tickets={tickets} sold={sold} total={total} />
    );
    const ticketsCard = wrapper.find("TicketsCard");
    expect(ticketsCard.prop("tickets")).toEqual(tickets);
    expect(ticketsCard.prop("sold")).toEqual(sold);
    expect(ticketsCard.prop("total")).toEqual(total);
  });

  it("renders the ViewsCard component", () => {
    const wrapper = shallow(
      <DashboardHeader tickets={tickets} sold={sold} total={total} />
    );
    expect(wrapper.find("ViewsCard")).toHaveLength(1);
  });

  it("does not render the Recommendations component", () => {
    const wrapper = shallow(
      <DashboardHeader tickets={tickets} sold={sold} total={total} />
    );
    expect(wrapper.find("Recommendations")).toHaveLength(0);
  });
});
