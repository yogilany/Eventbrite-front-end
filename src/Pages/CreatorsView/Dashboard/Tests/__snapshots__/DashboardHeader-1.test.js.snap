// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardHeader renders correctly 1`] = `ShallowWrapper {}`;
